### PR TITLE
introduced personal scripts dir

### PR DESCRIPTION
**Pull request by eprst**
`run_script` will now look for scripts under both `/scripts` and
`~/.${shell.name}/scripts`

### CODE REVIEW

File: shellbase-core/src/main/scala/com/sumologic/shellbase/ShellBase.scala [hunk 1]
```diff
@@ -86,6 +101,7 @@ abstract class ShellBase(val name: String) {
   /**
     * Name of the history file.
     */
+  //NOTE(Konstantin, 2017-27-02): This should probably be moved to $personalDir
```
Comment (weirded) [inline shellbase-core/src/main/scala/com/sumologic/shellbase/ShellBase.scala]: Nit: This format for a note won't pass scalastyle checks.

File: shellbase-core/src/main/scala/com/sumologic/shellbase/ShellBase.scala [hunk 2]
```diff
@@ -69,11 +69,26 @@ abstract class ShellBase(val name: String) {
...
 
   /**
+    * Where to look for personal scripts.
+    */
+  def personalScriptDir: File = new File(personalDir, "scripts/")
+
+  /**
+    * Start-up script
+    */
+  def initScript: File = new File(personalScriptDir, ".init." + scriptExtension)
```
[the remark this answers is not in the document]
Comment (weirded) [reply shellbase-core/src/main/scala/com/sumologic/shellbase/ShellBase.scala]: You could make this an `Option[File]`, filtering for `exists` and `canRead`

[the remark this answers is not in the document]
Comment (eprst) [reply shellbase-core/src/main/scala/com/sumologic/shellbase/ShellBase.scala]: good point

File: shellbase-core/src/main/scala/com/sumologic/shellbase/ShellBase.scala [hunk 3]
```diff
@@ -69,11 +69,26 @@ abstract class ShellBase(val name: String) {
   def banner: String = ""
 
   /**
+    * User personal directory with shell-related stuff.
+    */
+  def personalDir: File = new File(new File(System.getProperty("user.home")), "." + name)
```
[the remark this answers is not in the document]
Comment (eprst) [reply shellbase-core/src/main/scala/com/sumologic/shellbase/ShellBase.scala]: I think this should actually be the name. We probably don't want different shells with the same extension to share init files.

**Pull request merged**